test/oqs_test_hashnsign.c

```diff
@@ -0,0 +1,201 @@
+        md = EVP_sha3_256();
+    else if (strstr(lowercase_alg_name, "sha3-384"))
+        md = EVP_sha3_384();
+    else if (strstr(lowercase_alg_name, "sha3-512"))
+        md = EVP_sha3_512();
+    else if (strstr(lowercase_alg_name, "shake128"))
+        md = EVP_shake128();
+    else if (strstr(lowercase_alg_name, "shake256"))
+        md = EVP_shake256();
+
+    free(lowercase_alg_name);
+    return md;
+}
+
+static int test_hash_n_sign(const char *sigalg_name)
+{
+    EVP_MD_CTX *mdctx = NULL;
+    EVP_PKEY_CTX *pkey_ctx = NULL;
+    EVP_PKEY *key = NULL;
+    const char msg[] = "The quick brown fox jumps over the lazy dog";
+    unsigned char *sig = NULL;
+    size_t siglen;
+    int testresult = 0;
+
+    if (!alg_is_enabled(sigalg_name)) {
+        fprintf(stderr, "Not testing disabled algorithm %s.\n", sigalg_name);
+        return 1;
+    }
+
+    const EVP_MD *md_type = get_digest_for_algorithm(sigalg_name);
+    if (!md_type) {
+        fprintf(
+            stderr,
+            "Unsupported digest type for algorithm %s.\n Not failing over unsupported hash algs.",
+            sigalg_name);
+        return 1;
+    }
+
+    pkey_ctx = EVP_PKEY_CTX_new_from_name(libctx, sigalg_name, NULL);
+    if (!pkey_ctx) {
+        fprintf(stderr, "EVP_PKEY_CTX_new_from_name failed for %s.\n",
+                sigalg_name);
+        goto cleanup;
+    }
+
+    if (EVP_PKEY_keygen_init(pkey_ctx) <= 0) {
+        fprintf(stderr, "EVP_PKEY_keygen_init failed for %s.\n", sigalg_name);
+        goto cleanup;
+    }
+
+    if (EVP_PKEY_generate(pkey_ctx, &key) <= 0) {
+        fprintf(stderr, "EVP_PKEY_generate failed for %s.\n", sigalg_name);
+        goto cleanup;
+    }
+
+    mdctx = EVP_MD_CTX_new();
+    if (mdctx == NULL) {
+        fprintf(stderr, "EVP_MD_CTX_new failed for %s.\n", sigalg_name);
+        goto cleanup;
+    }
+
+    if (EVP_DigestSignInit(mdctx, NULL, md_type, NULL, key) <= 0) {
+        fprintf(stderr, "EVP_DigestSignInit failed for %s.\n", sigalg_name);
+        testresult = 0;
+        goto cleanup;
+    }
+
+    if (EVP_DigestSignUpdate(mdctx, msg, strlen(msg)) <= 0) {
+        fprintf(stderr, "EVP_DigestSignUpdate failed for %s.\n", sigalg_name);
+        testresult = 0;
+        goto cleanup;
+    }
+
+    if (EVP_DigestSignFinal(mdctx, NULL, &siglen) <= 0) {
+        fprintf(stderr, "EVP_DigestSignFinal (get length) failed for %s.\n",
+                sigalg_name);
+        testresult = 0;
+        goto cleanup;
+    }
+
+    sig = OPENSSL_malloc(siglen);
+    if (sig == NULL) {
+        fprintf(stderr, "OPENSSL_malloc failed for %s.\n", sigalg_name);
+        testresult = 0;
+        goto cleanup;
+    }
+
+    if (EVP_DigestSignFinal(mdctx, sig, &siglen) <= 0) {
+        fprintf(stderr, "EVP_DigestSignFinal (get signature) failed for %s.\n",
+                sigalg_name);
+        testresult = 0;
+        goto cleanup;
+    }
+
+    printf("Signature operation successful for %s.\n", sigalg_name);
```

Comment:
Is this unconditional `printf` intentional? Then we cannot do a silent (successful) test run, right?